includes/chatbot/components/buttons/class-omise-chatbot-component-button-orderstatus.php

```diff
@@ -0,0 +1,13 @@
+<?php
+defined( 'ABSPATH' ) or die( 'No direct script access allowed.' );
+
+if ( class_exists( 'Omise_Chatbot_Component_Button_Orderstatus' ) ) {
+	return;
+}
+
+class Omise_Chatbot_Component_Button_Orderstatus extends Omise_Chatbot_Component_Button_Postback {
```

Comment:
If we create button class and extend from button_type(postback) class, then if we have a new button we will create new file right ?
Can we just create a button_type class ex. Omise_Chatbot_Component_Button_Postback and pass data to constructor instead create a new file, like
`$featured_product_button = new Omise_Chatbot_Component_Button_Postback("title", "payload")`
What do you think ?

Reply:
@igroomgrim Sure, I think we should provide that too. (I mean, have both, specific classes for some often-used buttons that we can just easily reuse them and also have a way to create a custom button like you mentioned 👍? )

rule is: 
- if we use the same button more than 1 place, = have its own specific class.
- if we use only one time, just go with 
```
$featured_product_button = new Omise_Chatbot_Component_Button_Postback("title", "payload")
```

is that ok? or you think better to just have only `Omise_Chatbot_Component_Button_Postback` class without those `*_Orderstatus`, `*_Featuredproducts`, `*_Productcategory` prebuilt classes?

Reply:
I just see these class is look similar attribute and constructor (*_Orderstatus, *_Featuredproducts, *_Productcategory).
Then if we just only have Omise_Chatbot_Component_Button_Postback class it will be ok na.
🤖

Reply:
@igroomgrim i c. I have these 3 classes (`Omise_Chatbot_Component_Button_*`) to avoid below use case.
<img width="863" alt="screen shot 2560-09-25 at 11 17 55 pm" src="https://user-images.githubusercontent.com/2154669/30819070-cb7d9ea4-a247-11e7-8502-14ea0840eec0.png">

So we can refactor to
<img width="381" alt="screen shot 2560-09-25 at 11 24 14 pm" src="https://user-images.githubusercontent.com/2154669/30819366-ae0a13ec-a248-11e7-8378-ca6a45732ab3.png">

That we can reduce mistake from human-error (like, change `label` / `payload` at 1 place, but forgot to update another places).

Anyway, so far I saw these buttons have been used just only 2 places. So I think it's ok to go with only `Omise_Chatbot_Component_Button_Postback` for now unless we have more place that need to reuse those buttons.

👆 so, 
```php
$this->components['template_button']
	->set_text( 'I\'m glad you visited How may I assist ?' )
	->add_buttons(
		array(
			new Omise_Chatbot_Component_Button_Postback( __( 'Featured products', 'omise' ), Omise_Chatbot_Payloads::ACTION_FEATURED_PRODUCTS ),
			new Omise_Chatbot_Component_Button_Postback( __( 'Product category', 'omise' ), Omise_Chatbot_Payloads::ACTION_PRODUCT_CATEGORY ),
			new Omise_Chatbot_Component_Button_Postback( __( 'Check order status', 'omise' ), Omise_Chatbot_Payloads::ACTION_ORDER_STATUS ),
		)
	);
```
👌  ?

Reply:
btw, I still cheer to have its own class. 😂 
Because those `Omise_Chatbot_Facebook_Webhook_Event_Handler` classes don't need to taking care on `button title` or `action payload` at all, just initiate the right class and let's those button classes take care its own `title` and `payload`.

Reply:
Ohh after i see this 
<img width="381" alt="screen shot 2560-09-25 at 11 24 14 pm" src="https://user-images.githubusercontent.com/2154669/30819366-ae0a13ec-a248-11e7-8378-ca6a45732ab3.png">
it look clean and save my eye 🤣 .
Then it ok if we separate to more classssss. 
:bowtie:

Reply:
lol so no change now? 😁 

btw, ideally 👇 (if WordPress will go with `namespace` approach someday in the future)
<img width="391" alt="screen shot 2560-09-25 at 11 41 19 pm" src="https://user-images.githubusercontent.com/2154669/30820079-0f913dbe-a24b-11e7-9d01-c40a13f47842.png">

Reply:
`someday in the future` :tableflip:
It look very cool and cleannn. T.T